packages/core/content-releases/admin/src/components/EntryValidationPopover.tsx

```diff
@@ -0,0 +1,376 @@
+  requiredStage?: Stage;
+  entryStage?: Stage;
+}) => {
+  if (!contentTypeHasReviewWorkflow) {
+    return <CheckCircle fill="neutral200" />;
+  }
+  if (requiredStage && requiredStage.id !== entryStage?.id) {
+    return <CrossCircle fill="danger600" />;
+  }
+  return <CheckCircle fill="success600" />;
+};
+
+const getReviewStageMessage = ({
+  contentTypeHasReviewWorkflow,
+  requiredStage,
+  entryStage,
+  formatMessage,
+}: {
+  contentTypeHasReviewWorkflow: boolean;
+  requiredStage?: Stage;
+  entryStage?: Stage;
+  formatMessage: (messageDescriptor: MessageDescriptor, values?: Record<string, string>) => string;
+}) => {
+  if (!contentTypeHasReviewWorkflow) {
+    return formatMessage({
+      id: 'content-releases.pages.ReleaseDetails.entry-validation.review-stage.not-enabled',
+      defaultMessage: 'This entry is not associated to any workflow.',
+    });
+  }
+
+  if (requiredStage && requiredStage.id !== entryStage?.id) {
+    return formatMessage(
+      {
+        id: 'content-releases.pages.ReleaseDetails.entry-validation.review-stage.not-ready',
+        defaultMessage: 'This entry is not at the required stage for publishing. ({stageName})',
+      },
+      {
+        stageName: requiredStage?.name ?? '',
+      }
+    );
+  }
+  return formatMessage(
+    {
+      id: 'content-releases.pages.ReleaseDetails.entry-validation.review-stage.ready',
+      defaultMessage: 'This entry is at the required stage for publishing. ({stageName})',
+    },
+    {
+      stageName: requiredStage?.name ?? '',
+    }
+  );
+};
+
+const ReviewStageValidation = ({
+  contentTypeHasReviewWorkflow,
+  requiredStage,
+  entryStage,
+}: {
+  contentTypeHasReviewWorkflow: boolean;
+  requiredStage?: Stage;
+  entryStage?: Stage;
+}) => {
+  const { formatMessage } = useIntl();
+
+  const Icon = getReviewStageIcon({
+    contentTypeHasReviewWorkflow,
+    requiredStage,
+    entryStage,
+  });
+
+  return (
+    <Flex direction="column" gap={1} width="100%" padding={5}>
+      <Flex gap={2} width="100%">
+        <Typography fontWeight="bold">
+          {formatMessage({
+            id: 'content-releases.pages.ReleaseDetails.entry-validation.review-stage',
+            defaultMessage: 'Review stage',
+          })}
+        </Typography>
+        {Icon}
+      </Flex>
+      <Typography>
+        {getReviewStageMessage({
+          contentTypeHasReviewWorkflow,
+          requiredStage,
+          entryStage,
+          formatMessage,
+        })}
+      </Typography>
+    </Flex>
+  );
+};
+
+export const EntryValidationPopover = ({
+  action,
+  schema,
+  entry,
+  status,
+}: EntryValidationPopoverProps) => {
+  const { validate, isLoading } = unstable_useDocument(
+    {
+      collectionType: schema?.kind ?? '',
+      model: schema?.uid ?? '',
+    },
+    {
+      // useDocument makes a request to get more data about the entry, but we only want to have the validation function so we skip the request
+      skip: true,
+    }
```

Comment:
non related question, would we benefit of discussing refactoring the useDocument hook? the skip flag looks like a shortcut that makes the hook a bit more difficult to understand

Reply:
I agree we should find what is wrong in this hook, because there are multiple places where you need to use it but is more complicated than other thing. In my opinion it should be split into two separate hooks: `useDocument` to get data about the document and `useDocumentMetadata` (or something like that) to get data about the document or utils functions about the document (for example the schema, the validation fn, etc...)